Python/Product/Analysis/Interpreter/Ast/AstPythonInterpreterFactory.cs

```diff
@@ -0,0 +1,179 @@
+﻿// Python Tools for Visual Studio
+// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using Microsoft.PythonTools.Analysis;
+using Microsoft.PythonTools.Infrastructure;
+using Microsoft.PythonTools.Parsing;
+
+namespace Microsoft.PythonTools.Interpreter.Ast {
+    class AstPythonInterpreterFactory : IPythonInterpreterFactory {
+        private readonly string _databasePath;
+        private readonly object _searchPathsLock = new object();
+        private PythonLibraryPath[] _searchPaths;
+        private IReadOnlyDictionary<string, string> _searchPathPackages;
+
+        public AstPythonInterpreterFactory(
+            InterpreterConfiguration config,
+            InterpreterFactoryCreationOptions options
+        ) {
+            if (config == null) {
+                throw new ArgumentNullException(nameof(config));
+            }
+            if (options == null) {
+                options = new InterpreterFactoryCreationOptions();
+            }
+            Configuration = config;
+            LanguageVersion = Configuration.Version.ToLanguageVersion();
+
+            _databasePath = options.DatabasePath;
+
+            if (!GlobalInterpreterOptions.SuppressPackageManagers) {
+                try {
+                    var pm = options.PackageManager;
+                    if (pm != null) {
+                        pm.SetInterpreterFactory(this);
+                        pm.InstalledFilesChanged += PackageManager_InstalledFilesChanged;
```

Comment:
There's no unregistration for this event handler that I can see. Maybe this class needs to be disposable.